src/app/core/store/customer/basket/basket.reducer.ts

```diff
@@ -193,13 +200,31 @@ export const basketReducer = createReducer(
       submittedBasket: undefined,
     };
   }),
-  on(updateBasketItemsSuccess, deleteBasketItemSuccess, (state, action) => ({
+  on(updateBasketItemSuccess, (state, action) => ({
     ...state,
+    basket: {
+      ...state.basket,
+      lineItems: state.basket.lineItems.map(item =>
+        item.id === action.payload.lineItem.id ? action.payload.lineItem : item
+      ),
+    },
+    info: action.payload.info,
+    validationResults: initialValidationResults,
+  })),
+  on(updateBasketItemsSuccess, (state, action) => ({
+    ...state,
+    info: action.payload.info,
+    validationResults: initialValidationResults,
+  })),
+  on(deleteBasketItemSuccess, (state, action) => ({
+    ...state,
+    basket: { ...state.basket, lineItems: state.basket.lineItems.filter(item => item.id !== action.payload.itemId) },
     info: action.payload.info,
     validationResults: initialValidationResults,
   })),
   on(addItemsToBasketSuccess, (state, action) => ({
     ...state,
+    basket: { ...state.basket, lineItems: unionBy(action.payload.lineItems, state.basket.lineItems || [], 'id') },
```

Comment:
It's not going to make a difference here, but I think we should use `||` and `??` in a way that reflects our intentions:
- `??` when you want to catch an undefined or null value
- `||` when you want to use the second value if the first is falsy.